R/fread.R

```diff
@@ -41,6 +41,8 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="")
     } else {
       # avoid creating a tempfile() for single strings, which can be done a lot; e.g. in the test suite.
       input = text
+      if (!length(grep("\\n|\\r", input)))
```

Comment:
I would link the issue in comment so it will be easier to understand that code in future